collectors/o365/o365_collector.js

```diff
@@ -104,7 +106,7 @@ class O365Collector extends PawsCollector {
         if(moment().diff(state.since, 'days', true) > 7){
             const newStart = moment().subtract(PARTIAL_WEEK_HOURS, 'hours');
             state.since = newStart.toISOString();
-            state.until = newStart.add(1, 'hours').toISOString();
+            state.until = (process.env.paws_collection_interval && process.env.paws_collection_interval > 0) ? newStart.add(process.env.paws_collection_interval, 'seconds').toISOString() : newStart.add(1, 'hours').toISOString();
```

Comment:
let's simply overwrite `state` with new `until` value if `paws_collection_interval` is defined. And then we don't need any other changes below.